cmd/glasskube/cmd/repo_options.go

```diff
@@ -52,17 +60,72 @@ func (opts *repoOptions) BindToCmdFlags(cmd *cobra.Command) {
 }
 
 func (opts *repoOptions) Normalize() error {
+	if len(opts.Url) > 0 {
+		if _, err := url.ParseRequestURI(opts.Url); err != nil {
+			return fmt.Errorf("use a valid URL for the package repository (got %v)", opts.Url)
+		}
+	}
+
 	if len(opts.Username) > 0 || len(opts.Password) > 0 {
-		if opts.Auth != repoAddNoAuth {
+		if opts.Auth != repoBasicAuth {
 			return fmt.Errorf("username/password only applies to basic authentication (got %v)", opts.Auth)
 		} else {
-			opts.Auth = repoAddBasicAuth
+			opts.Auth = repoBasicAuth
 		}
 	} else if len(opts.Token) > 0 {
-		if opts.Auth != repoAddNoAuth {
+		if opts.Auth != repoBearerAuth {
 			return fmt.Errorf("token only applies to bearer authentication (got %v)", opts.Auth)
 		} else {
-			opts.Auth = repoAddBearerAuth
+			opts.Auth = repoBearerAuth
+		}
+	}
```

Comment:
This breaks a currently supported input:
```
glasskube repo add foo http://bar --username admin
```
This input implies that the auth type is "basic", since only basic auth uses the username value. 

In contrast, the following input was previously unsupported:
```
glasskube repo add foo http://bar --auth basic --username admin
```
Though I think it would make sense to support that as well.

You can change the conditions to `opts.Auth != repoNoAuth && opts.Auth != repo.BasicAuth` or something like that.

Reply:
Done, thanks!